apm/src/app/welcome/navigation.component.html

```diff
@@ -0,0 +1,10 @@
+<nav class='navbar navbar-light bg-light'>
+    <a class='navbar-brand'>{{pageTitle}}</a>
+    <ul class='nav nav-pills'>
+      <li><a class='nav-link' routerLink='/welcome'>Home</a></li>
+      <li><a class='nav-link' routerLink='/products'>Product List</a></li>
+      <li *ngIf="!userService.isAuthenticated()"><a class='nav-link' routerLink='/login'>Login</a></li>
```

Comment:
same expression is evaluated 3 times. Let me know if you have read about ng-template and ng-container

Reply:
Not yet. I will update this after I read about ng-template and ng-container